### PR TITLE
Support boolean types

### PR DESCRIPTION
**Pull request by NatashaHull**
This is an update on https://github.com/18F/open-data-maker/pull/129 that resolves the merge conflict and adds tests.

### CODE REVIEW

**Comment by yozlet** [comment]
@NatashaHull thanks for this! Is it still WIP or is it ready to go?

File: lib/data_magic/document_builder.rb [hunk 1]
```diff
@@ -90,6 +92,17 @@ def fix_field_type(type, value, key=nil)
...
       end
 
+      def parse_boolean(value)
+        case value
+        when "true"
+          true
+        when "false"
+          false
+        else
+          fail InvalidDictionary, "unexpected value '#{value.inspect}' for type '#{type.inspect}' for field '#{key}'"
```
Comment (yozlet) [inline lib/data_magic/document_builder.rb]: It's fine to add this here as a fallback, but the primary place to do this type-checking is in `DataMagic::ErrorChecker.report_wrong_field_type()` . I recommend expanding the code there to handle booleans.

**Comment by NatashaHull** [comment]
@yozlet This is basically ready to go. I left it was a wip in case someone wanted to add extra ways to refer to booleans.

File: lib/data_magic/document_builder.rb (hunk 1 above)
Comment (NatashaHull) [inline lib/data_magic/document_builder.rb]: Oh nice. I didn't notice that when I was in the code (for a few hours several Thursdays ago). Feel free to update this as necessary.

**Comment by ultrasaurus** [comment]
sorry for the long delay -- thanks for working on this!

**Pull request merged**